src/parachain/asset-registry.ts

```diff
@@ -0,0 +1,68 @@
+import { Currency, UnitList } from "@interlay/monetary-js";
+import { ApiPromise } from "@polkadot/api";
+import { StorageKey, u32 } from "@polkadot/types";
+import { OrmlAssetRegistryAssetMetadata } from "@polkadot/types/lookup";
+import { stripHexPrefix } from "../utils";
+import { Option } from "@polkadot/types-codec";
+
+/**
+ * @category BTC Bridge
+ */
+export interface AssetRegistryAPI {
+    /**
+     * Get all currencies (foreign assets) in the asset registry.
+     * @returns A list of currencies.
+     */
+    getForeignAssetsAsCurrencies(): Promise<Array<Currency<UnitList>>>;
+}
+
+// shorthand type for the unwieldy tuple
+export type AssetRegistryMetadataTuple = [StorageKey<[u32]>, Option<OrmlAssetRegistryAssetMetadata>];
+ 
+export class DefaultAssetRegistryAPI {
+
+    constructor(private api: ApiPromise) { }
+
+    // not private for easier testing
+    metadataToCurrency(metadata: OrmlAssetRegistryAssetMetadata): Currency<UnitList> {
```

Comment:
It makes more sense to define this either as a `static` method or as a utility function since it doesn't make use of any `DefaultAssetRegistryAPI` class property.